rasa/core/tracker_store.py

```diff
@@ -720,6 +723,95 @@ def keys(self) -> Iterable[Text]:
         return [c["sender_id"] for c in self.conversations.find()]
 
 
+class FirebaseRealtimeTrackerStore(TrackerStore):
+    """Stores conversation history in Firebase Realtime Database
+
+    Args:
+        db_root : root branch name to store conversations
+        firebase_key_file_path : path to firebase admin sdk key file
+
+    """
+
+    def __init__(
+        self,
+        domain: Domain,
+        firebase_key_file_path: Text,
+        db_storage_path: Text,
+        database_url: Text,
+        event_broker: Optional[EndpointConfig] = None,
+        **kwargs: Dict[Text, Any],
+    ) -> None:
+
+        self.store = {}
+        self.key_file_path = firebase_key_file_path
+        self.database_url = database_url
+        self.db_storage_path = db_storage_path
+
+        self.cred = credentials.Certificate(self.key_file_path)
+        init_app = firebase_admin.initialize_app(
+            self.cred, {"databaseURL": self.database_url}
+        )
+        super().__init__(domain, event_broker, **kwargs)
+
+    def save(self, tracker: DialogueStateTracker) -> None:
+        if self.event_broker:
+            self.stream_events(tracker)
+
+        serialised_tracker = FirebaseRealtimeTrackerStore.serialise_tracker(tracker)
+
+        user_message_id = f"user_{int(time())}"
+        user_message = tracker.latest_message.text
+
+        bot_message_id = f"bot_{int(time())}"
+        bot_message = self.get_latest_bot_message(serialised_tracker)
+
+        if bot_message != None:
+            self.push_conversation_db(
+                ref_path=f"{self.db_storage_path}/{tracker.sender_id}",
+                message_id=user_message_id,
+                message=user_message,
```

Comment:
This method call should be extracted outside this if statement, by checking first that `message_text is not None`, because regardless of whether there is a bot response, the user message should be saved, right?